src/main/java/io/pakland/mdas/githubstats/domain/model/Team.java

```diff
@@ -18,19 +17,78 @@
 public class Team {
 
   @Id
-  @Column(name = "id", updatable = false, nullable = false)
+  @Column(updatable = false, nullable = false)
   @GeneratedValue(strategy = GenerationType.IDENTITY)
-  private int id;
+  private Long id;
 
-  @Column(name = "organization_id")
-  private int organizationId;
-
-  @Column(name = "slug")
   private String slug;
 
   @Column(name = "member_url")
   private String memberUrl;
 
-  @Column(name = "repository_url")
-  private String repositoryUrl;
+  @ManyToOne(fetch = FetchType.LAZY)
+  private Organization organization;
+
+  @OneToMany(
+    mappedBy = "team",
+    cascade = CascadeType.ALL,
+    orphanRemoval = true
+  )
+  private List<User> users = new ArrayList<>();
+
+  @OneToMany(
+    mappedBy = "team",
+    cascade = CascadeType.ALL,
+    orphanRemoval = true
+  )
+  private List<Repository> repositories = new ArrayList<>();
+
+  @OneToMany(
+    mappedBy = "team",
+    cascade = CascadeType.ALL,
+    orphanRemoval = true
+  )
+  private List<HistoricQueries> querys = new ArrayList<>();
+
+  public void addUser(User user) {
+    users.add(user);
+    user.setTeam(this);
+  }
+
+  public void removeUser(User user) {
+    users.remove(user);
+    user.setTeam(null);
+  }
+
+  public void addRepository(Repository repository) {
+    repositories.add(repository);
+    repository.setTeam(this);
+  }
+
+  public void removeRepository(Repository repository) {
+    repositories.remove(repository);
+    repository.setTeam(null);
+  }
+
+  public void addQuerys(HistoricQueries historicQuery) {
+    querys.add(historicQuery);
+    historicQuery.setTeam(this);
+  }
+
+  public void removeQuerys(HistoricQueries historicQuery) {
+    querys.remove(historicQuery);
+    historicQuery.setTeam(null);
+  }
```

Comment:
Just a typo, since we are only adding one it should be singular: `addHistoricQuery` and `removeHistoricQuery`.  
Adding `Historic` name into it adds also more information.